src/Common/src/System/Security/Cryptography/CryptoPool.cs

```diff
@@ -13,6 +13,14 @@ internal static class CryptoPool
 
         internal static byte[] Rent(int minimumLength) => ArrayPool<byte>.Shared.Rent(minimumLength);
 
+        internal static void Return(ArraySegment<byte> arraySegment)
+        {
+            Debug.Assert(arraySegment.Array != null);
+            Debug.Assert(arraySegment.Offset == 0);
```

Comment:
It's fine if it maps to all uses, but I'm curious why you went with such an assert rather than just clearing the relevant region?

Reply:
So it could defer to the other method, which already assumes zero as the offset (overload as a pattern accelerator).  If we ever need the non-zero variant the assert will trip and we'll just have to change the code. (Yay internal methods :))